micro-rdk/src/esp32/board.rs

```diff
@@ -365,4 +372,36 @@ impl Board for EspBoard {
         }
         Err(BoardError::GpioPinError(pin as u32, "not configured"))
     }
+
+    fn add_digital_interrupt_callback(
+        &mut self,
+        pin: i32,
+        intr_type: InterruptType,
+        cb: gpio_isr_t,
+        arg: Option<*mut core::ffi::c_void>,
+    ) -> Result<(), BoardError> {
+        let p = self
```

Comment:
doesn't it check if the pin was configured as a digital interrupt pin? does it matter?

Reply:
so it means that we can register any pin as interrupt as long as they were in the pin array attribute at least?

Reply:
yes. it should fail when making the esp-idf interrupt handler [if the pin is not GPIO](https://github.com/espressif/esp-idf/blob/f9765d03162efd48feece14a4949de66356e00ae/components/esp_driver_gpio/src/gpio.c#L554). 

I just came across a warning that pins 36 and 39 shouldn't be used as an interrupt, so added a check for those.